app/src/main/java/org/mifos/mobile/viewModels/ReviewLoanApplicationViewModel.kt

```diff
@@ -63,14 +67,24 @@ class ReviewLoanApplicationViewModel @Inject constructor(var dataManager: DataMa
 
     fun getLoanState() = loanState
 
-    fun submitLoan(): Observable<ResponseBody?>? {
+    private val _submitLoanUiState = MutableLiveData<SubmitLoanUiState>()
+    val submitLoanUiState: LiveData<SubmitLoanUiState> = _submitLoanUiState
+
+    fun submitLoan() = viewModelScope.launch(Dispatchers.IO) {
         loansPayload.productName = null
         loansPayload.loanPurpose = null
         loansPayload.currency = null
-        return if (loanState == LoanState.CREATE) {
-            dataManager?.createLoansAccount(loansPayload)
-        } else {
-            dataManager?.updateLoanAccount(loanId, loansPayload)
+
+        _submitLoanUiState.postValue(SubmitLoanUiState.Loading)
+        try {
+            val response = if (loanState == LoanState.CREATE) {
+                dataManager?.createLoansAccount(loansPayload)
+            } else {
+                dataManager?.updateLoanAccount(loanId, loansPayload)
+            }
+            _submitLoanUiState.postValue(SubmitLoanUiState.Next(response!!))
```

Comment:
When you use StateFlow you wont be using postValue because there is no set nor post. It’s just the stateFlow.value.
There is a difference between postValue and value when using LiveData , you can read it from the internet .